services/gitlab/gitlab-issues.service.js

```diff
@@ -0,0 +1,446 @@
+      },
+      documentation: customDocumentation + labelDocumentation,
+    },
+    {
+      title: 'GitLab (self-managed) closed issues by-label',
+      pattern: 'closed-raw/:project+',
+      namedParams: {
+        project: 'gitlab-cn/gitlab',
+      },
+      queryParams: {
+        gitlab_url: 'https://jihulab.com',
+        labels: 'backend,feature::enhancement',
+      },
+      staticPreview: {
+        label: 'closed backend,feature::enhancement issues',
+        message: '57',
+        color: 'yellow',
+      },
+      documentation: customDocumentation + labelDocumentation,
+    },
+    {
+      title: 'GitLab (self-managed) all issue',
+      pattern: 'all/:project+',
+      namedParams: {
+        project: 'gitlab-cn/gitlab',
+      },
+      queryParams: { gitlab_url: 'https://jihulab.com' },
+      staticPreview: {
+        label: 'issues',
+        message: '969 all',
+        color: 'yellow',
+      },
+      documentation: customDocumentation,
+    },
+    {
+      title: 'GitLab (self-managed) all issue',
+      pattern: 'all-raw/:project+',
+      namedParams: {
+        project: 'gitlab-cn/gitlab',
+      },
+      queryParams: { gitlab_url: 'https://jihulab.com' },
+      staticPreview: {
+        label: 'all issues',
+        message: '969',
+        color: 'yellow',
+      },
+      documentation: customDocumentation,
+    },
+    {
+      title: 'GitLab (self-managed) all issues by-label',
+      pattern: 'all-raw/:project+',
+      namedParams: {
+        project: 'gitlab-cn/gitlab',
+      },
+      queryParams: {
+        gitlab_url: 'https://jihulab.com',
+        labels: 'backend,feature::enhancement',
+      },
+      staticPreview: {
+        label: 'all backend,feature::enhancement issues',
+        message: '84',
+        color: 'yellow',
+      },
+      documentation: customDocumentation + labelDocumentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'issues', color: 'informational' }
+
+  static render({ variant, raw, statistics, labels }) {
+    let state = variant
+    if (state === 'opened') {
+      state = 'open'
+    }
+    const isMultiLabel = labels && labels.includes(',')
+    const labelText = labels ? `${isMultiLabel ? `${labels}` : labels} ` : ''
+
+    let labelPrefix = ''
+    let messageSuffix = ''
+    if (raw !== undefined) {
+      labelPrefix = `${state} `
+    } else {
+      messageSuffix = state
+    }
+
+    let issueCount
+    switch (state) {
+      case 'open':
+        issueCount = statistics.counts.opened
+        break
+      case 'closed':
+        issueCount = statistics.counts.closed
+        break
+      case 'all':
+        issueCount = statistics.counts.all
+        break
+    }
+    const renderLabel = `${labelPrefix}${labelText}issues`
+
+    return {
+      label: renderLabel,
+      message: `${metric(issueCount)}${
+        messageSuffix ? ' ' : ''
+      }${messageSuffix}`,
+      color: issueCount > 0 ? 'yellow' : 'brightgreen',
+    }
+  }
```

Comment:
I appreciate that you're (presumably) trying to model this, but we can deviate in the implementation to account for the specific nature of the upstream service and response structure. Particularly here I feel like we're conflating the response processing and rendering concerns in a manner that's not necessary and adds complexity.

For example, the entire process deriving the count of issues should be moved to a `transform` function, with the result simply being passed as an argument to this function